cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java

```diff
@@ -113,32 +277,138 @@ public Builder set(String fieldName, @Nullable Object value) {
       return this;
     }
 
+    /**
+     * Sets the date value for {@link Schema.LogicalType#DATE} field
+     *
+     * @param fieldName name of the field to set
+     * @param localDate value for the field
+     * @return this builder
+     * @throws UnexpectedFormatException if the field is not in the schema, or the field is not nullable but a null
+     *                                   value is given
+     * @throws IllegalArgumentException if the provided date is an invalid date
+     */
+    public Builder setDate(String fieldName, @Nullable LocalDate localDate) {
+      Schema.Field field = validateAndGetField(fieldName, localDate);
```

Comment:
This method should be `createField` (which will do validation as part of the creation). It's not good to name a method based on the implementation.